### DISCUSSION TITLE
modal editing of prompt text

### DISCUSSION THREAD

**Discussion by Zoybean** [helix-editor/helix#10331] [Ideas]
Sometimes, a command or other prompt (e.g. rename) will require typing or editing text using readline, where I cannot use helix's modal editing techniques. Is there any chance we could get the ability to edit text in prompts using modal editing, as if the prompt were its own buffer? I figure this could be opted into via some chord (ideally one that wouldn't already mean something in readline). 

example use-case: (using ctrl-o as the mode-change chord)
- I want to rename a symbol to change to lower-case (specifically using the LSP's symbol rename function)
Currently, I have to: `space` `r` (then delete / re-type the name) `enter` 
I would prefer to: `space` `r` `ctrl-o` `B` `` ` `` `enter`

Complications and open questions:
- how to exit this modal editing sub-mode (whether confirming or cancelling the prompt)? Treating it like a buffer and using `:x` to finish editing, could be confusing and would possibly introduce recursion with this command sub-mode (which could have its own edit sub-mode). Above, I've just used `enter`, as I don't feel prepared to answer these questions myself.
- Ideally (for me) this would work in most prompts (search prompt, rename prompt, command prompt) but I'm not sure about the picker-style prompts (e.g. file picker, command palette, global search results).

**Comment by woojiq** [comment]
~FYI, you can use emacs/bash keybinds inside prompt. For example, https://www.pythian.com/blog/technical-track/emacs-keybindings-in-bash.
Personally, I don't think helix modal editing inside a prompt is the good idea.~
Nevermind (misread what you are trying to achieve)
